plugin.video.retrospect/resources/libs/chn_class.py

```diff
@@ -0,0 +1,1053 @@
+        :rtype: MediaItem
+
+        """
+
+        Logger.debug('Starting update_video_item for %s (%s)', item.name, self.channelName)
+
+        data = UriHandler.open(item.url, proxy=self.proxy, additional_headers=item.HttpHeaders)
+
+        url = Regexer.do_regex(self.mediaUrlRegex, data)[-1]
+        part = MediaItemPart(item.name, url)
+        item.MediaItemParts.append(part)
+
+        Logger.info('finishing update_video_item. MediaItems are %s', item)
+
+        if not item.thumb and self.noImage:
+            # no thumb was set yet and no url
+            Logger.debug("Setting thumb to %s", item.thumb)
+            item.thumb = self.noImage
+
+        if not item.has_media_item_parts():
+            item.complete = False
+        else:
+            item.complete = True
+        return item
+
+    def log_on(self):
+        """ Logs on to a website, using an url.
+
+        First checks if the channel requires log on. If so and it's not already
+        logged on, it should handle the log on. That part should be implemented
+        by the specific channel.
+
+        More arguments can be passed on, but must be handled by custom code.
+
+        After a successful log on the self.loggedOn property is set to True and
+        True is returned.
+
+        :return: indication if the login was successful.
+        :rtype: bool
+
+        """
+
+        if not self.requiresLogon:
+            Logger.debug("No login required of %s", self.channelName)
+            return True
+
+        if self.loggedOn:
+            Logger.info("Already logged in")
+            return True
+
+        return False
+
+    def play_video_item(self, item, bitrate=None):
+        """ Starts the playback of the <item> with the specific <bitrate> in the selected <player>.
+
+        Starts the playback of the selected MediaItem <item>. Before playback is started
+        the item is check for completion (item.complete), if not completed, the self.update_video_item
+        method is called to update the item.
+
+        After updating the requested bitrate playlist is selected, if bitrate was set to None
+        the bitrate is retrieved from the addon settings. The playlist is then played using the
+        requested player.
+
+        :param MediaItem item:      The item to start playing
+        :param int|None bitrate:    The requested bitrate in Kbps or None.
+
+        :return: A Kodi Playlist for this MediaItem and a subtitle.
+        :rtype: xbmc.PlayList
+
+        """
+
+        if bitrate is None:
+            # use the bitrate from the Kodi settings if bitrate was not specified and the item is MultiBitrate
+            bitrate = AddonSettings.get_max_stream_bitrate(self)
+
+        # get the playlist
+        play_list = item.get_kodi_play_list(bitrate, update_item_urls=True, proxy=self.proxy)
+
+        # call for statistics with timing
+        Statistics.register_playback(self, item, Initializer.StartTime)
```

Comment:
what is this?

Reply:
It registered a playback of an item with Google Analytics. I used GA to see if a channel broke due to serverside changes. In that case the playback count dropped to nearly 0. 

Removed this.